pspipe_utils/consistency.py

```diff
@@ -21,33 +21,45 @@ def append_spectra_and_cov(ps_dict,
     ----------
     ps_dict: dict
       dict containing the different power spectra
-    cov_list: dict
+      if set to None, will only combine the covariances
+    cov_dict: dict
       dict containing the covariances
+      if set to None, will only combine the spectra
     spec_list: list
       list containing the name of the spectra
 
     """
+
+    cov_only = (ps_dict is None) and (cov_dict is not None)
+    ps_only = (cov_dict is None) and (ps_dict is not None)
+
+
     n_bins = len(ps_dict[spec_list[0]])
     n_spec = len(spec_list)
 
-    spectra_vec = np.zeros(n_bins * n_spec)
-    full_cov = np.zeros((n_bins * n_spec,
-                         n_bins * n_spec))
+    spectra_vec = None if cov_only else np.zeros(n_bins * n_spec)
+
+    full_cov = None if ps_only else np.zeros((n_bins * n_spec,
+                                              n_bins * n_spec))
 
     for i, key1 in enumerate(spec_list):
-        spectra_vec[i*n_bins:(i+1)*n_bins] = ps_dict[key1]
+        if not cov_only:
+            spectra_vec[i*n_bins:(i+1)*n_bins] = ps_dict[key1]
         for j, key2 in enumerate(spec_list):
             if j < i: continue
-            try:
-                full_cov[i*n_bins:(i+1)*n_bins,
-                         j*n_bins:(j+1)*n_bins] = cov_dict[key1, key2]
-            except:
-                full_cov[i*n_bins:(i+1)*n_bins,
-                         j*n_bins:(j+1)*n_bins] = cov_dict[key2, key1]
 
-    full_cov = np.triu(full_cov)
-    transpose_cov = full_cov.T
-    full_cov += transpose_cov - np.diag(full_cov.diagonal())
+            if not ps_only:
+                try:
+                    full_cov[i*n_bins:(i+1)*n_bins,
+                            j*n_bins:(j+1)*n_bins] = cov_dict[key1, key2]
+                except:
```

Comment:
to make sure to catch the right exception i.e. `KeyError` you should write `except KeyError:`